packages/battery_plus/battery_plus/android/src/main/kotlin/dev/fluttercommunity/plus/battery/BatteryPlusPlugin.kt

```diff
@@ -0,0 +1,195 @@
+import android.provider.Settings
+import androidx.annotation.RequiresApi
+
+/** BatteryPlusPlugin  */
+class BatteryPlusPlugin : MethodCallHandler, EventChannel.StreamHandler, FlutterPlugin {
+    private var applicationContext: Context? = null
+    private var chargingStateChangeReceiver: BroadcastReceiver? = null
+    private var methodChannel: MethodChannel? = null
+    private var eventChannel: EventChannel? = null
+
+    override fun onAttachedToEngine(binding: FlutterPluginBinding) {
+        this.applicationContext = binding.applicationContext
+        methodChannel = MethodChannel(binding.binaryMessenger, "dev.fluttercommunity.plus/battery")
+        eventChannel = EventChannel(binding.binaryMessenger, "dev.fluttercommunity.plus/charging")
+        eventChannel!!.setStreamHandler(this)
+        methodChannel!!.setMethodCallHandler(this)
+    }
+
+    override fun onDetachedFromEngine(binding: FlutterPluginBinding) {
+        applicationContext = null
+        methodChannel!!.setMethodCallHandler(null)
+        methodChannel = null
+        eventChannel!!.setStreamHandler(null)
+        eventChannel = null
+    }
+
+    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
+        when (call.method) {
+            "getBatteryLevel" -> {
+                val currentBatteryLevel = getBatteryLevel()
+                if (currentBatteryLevel != -1) {
+                    result.success(currentBatteryLevel)
+                } else {
+                    result.error("UNAVAILABLE", "Battery level not available.", null)
+                }
+            }
+            "getBatteryState" -> {
+                val currentBatteryStatus = getBatteryState()
+                if (currentBatteryStatus != null) {
+                    result.success(currentBatteryStatus)
+                } else {
+                    result.error("UNAVAILABLE", "Charging status not available.", null)
+                }
+            }
+            "isInBatterySaveMode" -> {
+                val isInPowerSaveMode = isInPowerSaveMode()
+                if (isInPowerSaveMode != null) {
+                    result.success(isInPowerSaveMode)
+                } else {
+                    result.error("UNAVAILABLE", "Battery save mode not available.", null)
+                }
+            }
+            else -> result.notImplemented()
+        }
+    }
+
+    @TargetApi(VERSION_CODES.O)
+    override fun onListen(arguments: Any, events: EventSink) {
+        chargingStateChangeReceiver = createChargingStateChangeReceiver(events)
+        applicationContext!!.registerReceiver(chargingStateChangeReceiver, IntentFilter(Intent.ACTION_BATTERY_CHANGED))
+        val status = getBatteryState()
+        publishBatteryStatus(events, status)
+    }
+
+    override fun onCancel(arguments: Any) {
+        applicationContext!!.unregisterReceiver(chargingStateChangeReceiver)
+        chargingStateChangeReceiver = null
+    }
+
+    private fun getBatteryState(): String? {
```

Comment:
Is there a reason, this was renamed from `getBatteryStatus()` to `getBatteryState()`?
In line 59 it says `val currentBatteryStatus = getBatteryState()`, the method call is named `getBatteryState`.
Could it be `state` or `status` (breaking) everywhere?

Reply:
Could we rename everything containing `batteryStatus` to `batteryState`?

Reply:
I have brought back the old name to this function for consistency with other functions and constants from the `BatteryManager` class, like `BatteryManager.BATTERY_STATUS_CHARGING`

Reply:
Sounds and looks great 🚀